chore(Android): Adding Android player example

casajavi opened the pull request.

casajavi committed:
chore(Android): Adding Android player example

lomudi commented:
@drorg @javier-tarazaga @guysarfatty can you review this? android is not my stong side... :)

lomudi commented:
@casajavi as an external developer, I will need to get any special permissions in order to run the example app? (like GitHub repo or Bintry)

casajavi commented:
@lomudi You know what I forgot to detail that part ... we do need to grant access to the Bintray so they have access to the SDK dependencies. That would happen upon request and we would give them read access. Not sure if that's something support can do for us but for now, it can go directly to us. Thoughts ??

lomudi commented:
@casajavi lets at first step add it to the doc - which permissions/access needed and let's write that "you should get it from your Solution Engineer at Applicaster". Next steps will be to check if we can make it open, without getting a specific access.

javier-tarazaga commented on VideoPlayer/Android/app/build.gradle:
```diff
@@ -0,0 +1,35 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 26
```
Move it to 27 to match the app

javier-tarazaga commented on VideoPlayer/Android/app/src/androidTest/java/com/applicaster/player/plugins/zapppluginplayerexampleandroid/ExampleInstrumentedTest.java:
```diff
@@ -0,0 +1,26 @@
+package com.applicaster.player.plugins.zapppluginplayerexampleandroid;
+
```
Do we need this for anything in a tutorial?

javier-tarazaga commented on VideoPlayer/Android/app/src/main/res/raw/plugin_configurations.json:
```diff
@@ -0,0 +1,26 @@
+    "unsupported_since_zapp_sdk": ""
+  },
+  "configuration_json": {
+    "accountID": "Z4d3kxOn0tfJK9oJkcCpNlNGK1aP"
```
What is this accountID for? 🤔

javier-tarazaga commented on VideoPlayer/Android/app/src/test/java/com/applicaster/player/plugins/zapppluginplayerexampleandroid/ExampleUnitTest.java:
```diff
@@ -0,0 +1,17 @@
+package com.applicaster.player.plugins.zapppluginplayerexampleandroid;
```
Do we need this at all for a sample project?

javier-tarazaga commented on VideoPlayer/Android/testplayerplugin-android/build.gradle:
```diff
@@ -0,0 +1,46 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 26
```
Move it up to 27

javier-tarazaga commented on VideoPlayer/Android/testplayerplugin-android/build.gradle:
```diff
@@ -0,0 +1,46 @@
+
+    implementation 'com.android.support:appcompat-v7:26.1.0'
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
+    testImplementation 'junit:junit:4.12'
```
Do we need to include test dependencies in an example project?

javier-tarazaga commented on VideoPlayer/Android/testplayerplugin-android/src/androidTest/java/com/applicaster/player/plugins/testplayerplugin_android/ExampleInstrumentedTest.java:
```diff
@@ -0,0 +1,26 @@
+package com.applicaster.player.plugins.testplayerplugin_android;
+
```
Do we need this at all?

javier-tarazaga commented on VideoPlayer/Android/testplayerplugin-android/src/main/java/com/applicaster/player/plugins/testplayerplugin_android/TestPlayerActivity.java:
```diff
@@ -0,0 +1,68 @@
+                @Override
+                public void onPrepared(MediaPlayer mediaPlayer) {
+                    mediaPlayer.start();
+                    mediaController = new MediaController(TestPlayerActivity.this);
```
Can this get called multiple times during a video playback?

casajavi replied:
yes it can

javier-tarazaga commented on VideoPlayer/Android/testplayerplugin-android/src/main/java/com/applicaster/player/plugins/testplayerplugin_android/TestPlayerAdapter.java:
```diff
@@ -0,0 +1,186 @@
+            videoView.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
+                @Override
+                public void onPrepared(MediaPlayer mediaPlayer) {
+                    mediaPlayer.start();
```
Same as before. Why do we need it in 2 different places?

casajavi replied:
One is for the inline, one is for the fullscreen

javier-tarazaga commented on VideoPlayer/Android/testplayerplugin-android/src/test/java/com/applicaster/player/plugins/testplayerplugin_android/ExampleUnitTest.java:
```diff
@@ -0,0 +1,17 @@
+package com.applicaster.player.plugins.testplayerplugin_android;
```
Do we need this at all?

javier-tarazaga reviewed (commented):
Here lays the fundamental question. Do we want to show external developers how a plugin is written in the real context of an app or simply showing how could it be done and then use the starter-kit project (which actually need updates) to actually develop their plugin?

casajavi committed:
chore(updates): Updates and improvements

casajavi commented:
@lomudi I have added the section detailing the Bintray access for our SDK dependencies on the README. @javier-tarazaga I fixed the open issues you commented on. Thanks guys!

lomudi reviewed (approved):
💯 
looks good.
the next step will be to write a step by step tutorial, also for iOS :)

lomudi commented:
@casajavi don't merge yet, wait also for @javier-tarazaga approval :)
